src/unstructured.rs

```diff
@@ -405,11 +405,9 @@ impl<'a> Unstructured<'a> {
     /// ```
     pub fn fill_buffer(&mut self, buffer: &mut [u8]) -> Result<()> {
         let n = std::cmp::min(buffer.len(), self.data.len());
-        for i in 0..n {
-            buffer[i] = self.data[i];
-        }
-        for i in self.data.len()..buffer.len() {
-            buffer[i] = 0;
+        buffer[..n].copy_from_slice(&self.data[..n]);
+        for byte in buffer[n..].iter_mut() {
+            *byte = 0;
```

Comment:
This is the only non obvious change. Clippy didn't suggest exactly this solution but it complained about the previous loop so I changed it like this. Imo this makes it more clear that the rest of the buffer is set to 0.